tests/queries/0_stateless/01157_replace_table.sql

```diff
@@ -0,0 +1,48 @@
+drop table if exists t;
+drop table if exists dist;
+drop table if exists buf;
+drop table if exists join;
+
+select 'test flush on replace';
+create table t (n UInt64, s String default 's' || toString(n)) engine=Memory;
+create table dist (n int) engine=Distributed(test_shard_localhost, currentDatabase(), t);
+create table buf (n int) engine=Buffer(currentDatabase(), dist, 1, 10, 100, 10, 100, 1000, 1000);
+
```

Comment:
To avoid distirubted sends in background, and explicitly check flush on REPLACE worth adding:
```suggestion
system stop distributed sends dist;
```